frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -1184,6 +1121,37 @@ export default ({
         e?.dataTransfer.files?.length &&
           this.$refs.sendArea.fileAttachmentHandler(e?.dataTransfer.files, true)
       }
+    },
+    async processSwitchQueue () {
+      if (this.ephemeral.chatroomSwitchQueue.length === 0) return
+
+      // Take the most recent chatroom entry on the queue and discard everything else. This way we can speed up the process of switching chatrooms.
+      const targetChatroomId = this.ephemeral.chatroomSwitchQueue.pop()
+      this.ephemeral.chatroomSwitchQueue = []
+      this.ephemeral.renderingChatRoomId = targetChatroomId
+
+      try {
+        await this.initializeState(true)
+        if (this.ephemeral.chatroomSwitchQueue.length > 0) {
+          // If the user has since switched to another chatroom while initializing this chatroom, stop here
+          // and care about the switched chatroom.
+          return this.processSwitchQueue()
+        } else {
+          this.ephemeral.messagesInitiated = false
+          this.ephemeral.unprocessedEvents = []
+          this.ephemeral.infiniteLoading?.reset()
```

Comment:
Grok says:

> this.ephemeral.infiniteLoading?.reset() in processSwitchQueue assumes the infinite-loading component is mounted, but it might not be if the component hasn’t rendered yet.

And suggests:

```js
this.$nextTick(() => this.$refs['infinite-loading']?.reset())
```

I don't know if this is a good idea or not, will leave it up to you. /cc @pieer  if you have thoughts

Reply:
This line used to be something like
```js
if (this..ephemeral.infiniteLoading) {
  this.ephemeral.infiniteLoading.reset()
}
```
Let me leave it as is, as this has been working fine like this.